### DISCUSSION TITLE
20. 대칭키와 비대칭키에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by eun-byeol** [jewelry-box/cs-interview#22] [General]
대칭키와 비대칭키에 대해 설명해 주세요.

**Comment by linirini** [comment]
## 대칭키 암호화
### 개념
- **하나의 키**를 사용하여 데이터를 암호화하고 복호화하는 방식
- 송신자와 수신자가 같은 키를 공유
### 작동 원리
1. 송신자가 대칭키로 데이터를 암호화하여 전송한다.
2. 수신자는 같은 키를 사용하여 암호화된 데이터를 복호화한다.
### 특징
- 연산이 단순하기 때문에 처리 속도가 빠르다.
- 같은 키를 사용하기 위해 키를 공유하는 과정이 필요하다. 그 과정에서 키가 노출되면 데이터의 암호화나 복호화가 가능해지므로 데이터 노출 위험이 존재한다.
  - 따라서 안전하게 키를 공유할 방법이 필요하다.
## 비대칭키 암호화
### 개념
- **서로 다른 두 개의 키**를 사용하여 데이터를 암호화하고 복호화한다.
- **공개키**: 누구나 볼 수 있게 공개된 키
- **비밀키(개인키)**: 본인만 소유하며 비공개로 유지
### 작동 원리
1. 송신자는 수신자의 **공개키**를 사용하여 데이터를 암호화한다.
2. 수신자는 자신만 가진 **개인키**로 데이터를 복호화한다.

** 디지털 서명
- 송신자가 개인키로 암호화하고, 수신자는 공개키로 복호화한다.
- 누가 암호화한 데이터인지 알 수 있기 때문에, 서명으로서 역할한다.
### 특징
- 암호화와 복호화에 사용되는 키가 다르므로, 연산이 복잡하여 상대적 느리다.
- 키를 공유할 필요가 없기 때문에 키 관리가 용이하다.

**Reply by linirini** [reply]
디지털 서명 참고
https://www.digicert.com/kr/faq/signature-trust/what-is-a-digital-signature

**Reply by eun-byeol** [reply]
> 서명자는 비공개 키로 해시를 암호화하는 반면, 읽는 이는 서명자의 공개 키로 암호를 해독합니다. 

오 덕분에 다시 이해했어요~😚
